### PR TITLE
Option to delete wallet

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
After redeploying a network, old wallet addresses and canister IDs can cause confusing errors.  So let's get rid of them.

# Changes
* Added flags to delete the wallet, canister IDs or both for a given network.

# Tests
## Test wallet deletion
 Set a wallet address for small06:
```
max@sinkpad:~/dfn/nns-dapp/branches/delete-network (3:18:54)$ DFX_NETWORK=small06 jq '.identities.default[env.DFX_NETWORK]="fubar"' "$WALLET_FILE" | tee /dev/stderr > , && mv , "$WALLET_FILE"
{
  "identities": {
    "default": {
      "nnsdapp-prod": "vo5te-2aaaa-aaaaa-aaazq-cai",
      "nnsdapp_both": "qvhpv-4qaaa-aaaaa-aaagq-cai",
      "small07": "fubar",
      "small11": "qsgjb-riaaa-aaaaa-aaaga-cai",
      "small11-prod": "qhbym-qaaaa-aaaaa-aaafq-cai",
      "small12": "q4eej-kyaaa-aaaaa-aaaha-cai",
      "svelte": "vh6yy-miaaa-aaaaa-aaaya-cai",
      "testnet": "qhbym-qaaaa-aaaaa-aaafq-cai",
      "small06": "fubar"
    }
  }
}
```
Delete it:
```
max@sinkpad:~/dfn/nns-dapp/branches/delete-network (3:18:56)$ ./deploy.sh --delete small06
{
  "config": {
    "BUILD": "dev",
    "FETCH_ROOT_KEY": true,
    "REDIRECT_TO_LEGACY": "both",
    "ENABLE_NEW_SPAWN_FEATURE": true,
    "HOST": "https://small06.dfinity.network",
    "IDENTITY_SERVICE_URL": "https://qjdve-lqaaa-aaaaa-aaaeq-cai.small06.dfinity.network"
  },
  "providers": [
    "http://[2a00:fb01:400:42:5000:caff:fed1:b2e7]:8080"
  ],
  "type": "persistent"
}

DELETE_CANISTER_IDS=true
DELETE_WALLET=true
START_DFX=false
DEPLOY_NNS_BACKEND=false
DEPLOY_II=false
DEPLOY_NNS_DAPP=false
POPULATE=false
OPEN_NNS_DAPP=false
Deleting the wallet for small06 in /home/max/.config/dfx/identity/default/wallets.json ...
```
Checking, and indeed small06 is gone.
```
max@sinkpad:~/dfn/nns-dapp/branches/delete-network (3:18:59)$ cat $WALLET_FILE 
{
  "identities": {
    "default": {
      "nnsdapp-prod": "vo5te-2aaaa-aaaaa-aaazq-cai",
      "nnsdapp_both": "qvhpv-4qaaa-aaaaa-aaagq-cai",
      "small07": "fubar",
      "small11": "qsgjb-riaaa-aaaaa-aaaga-cai",
      "small11-prod": "qhbym-qaaaa-aaaaa-aaafq-cai",
      "small12": "q4eej-kyaaa-aaaaa-aaaha-cai",
      "svelte": "vh6yy-miaaa-aaaaa-aaaya-cai",
      "testnet": "qhbym-qaaaa-aaaaa-aaafq-cai"
    }
  }
}
```
## Test canister ID deletion
This test is similar to wallet deletion.

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -48,6 +48,15 @@ help_text() {
...
 	  Print the steps that seem to be necessary for deployment.
 
+	--delete
+	  Delete network entries before deploying.
+
+	--delete-canister-ids
+	  Delete canister_id.json entries for the network before deploying.
+
+	--delete-wallet
+	  Delete wallet for the network before deploying.
```
Comment (lmuntaner) [inline deploy.sh:58]: what does this do? What is a "wallet" in this context?

**Review by lmuntaner — APPROVED** [review]
Just a question, but looks good!

I was doing it manually until now, hehe.

It might be interesting also to be able to delete only the nns-dapp canister id from canister_ids.json.

Not in this PR though, just a thuoght.

File: deploy.sh (hunk 1 above)
Comment (bitdivine) [reply deploy.sh:58]: It is the address of a cycles wallet.  Sample wallet file:
```
 $ cat /home/max/.config/dfx/identity/default/wallets.json
{
  "identities": {
    "default": {
      "nnsdapp-prod": "vo5te-2aaaa-aaaaa-aaazq-cai",
      "nnsdapp_both": "qvhpv-4qaaa-aaaaa-aaagq-cai",
      "small11": "qsgjb-riaaa-aaaaa-aaaga-cai",
      "small11-prod": "qhbym-qaaaa-aaaaa-aaafq-cai",
      "small12": "q4eej-kyaaa-aaaaa-aaaha-cai",
      "svelte": "vh6yy-miaaa-aaaaa-aaaya-cai",
      "testnet": "qhbym-qaaaa-aaaaa-aaafq-cai"
    }
  }
}
```

Comment (bitdivine) [reply deploy.sh:58]: For most of our work we don't need a wallet, we do for spawning an SNS.

Comment (lmuntaner) [reply deploy.sh:58]: ah, ok, thanks for the explanation!

**Pull request merged**